src/components/DocumentCard/DocumentCardLocation.tsx

```diff
@@ -1,22 +1,28 @@
 import * as React from 'react';
 import { IDocumentCardLocationProps } from './DocumentCard.Props';
 import './DocumentCardLocation.scss';
+import { autobind } from '../../utilities/autobind';
 
 export class DocumentCardLocation extends React.Component<IDocumentCardLocationProps, any> {
   public render() {
-    let { location, locationHref, onClick, ariaLabel } = this.props;
-
-    // If no onClick Function was provided and we do have an locationHref, create a function from it.
-    if (!onClick && locationHref) {
-      onClick = (ev: Event) => {
-        window.location.href = locationHref;
-        ev.preventDefault();
-        ev.stopPropagation();
-      };
-    }
+    let { location, locationHref, ariaLabel } = this.props;
 
     return (
-      <a className='ms-DocumentCardLocation' href={ locationHref } onClick={ onClick } aria-label={ ariaLabel }>{ location }</a>
+      <a className='ms-DocumentCardLocation' href={ locationHref } onClick={ this._onClick } aria-label={ ariaLabel }>{ location }</a>
     );
   }
+
+  @autobind
+  private _onClick(ev: React.MouseEvent): void {
+    let { locationHref, onClick } = this.props;
+
+    if (onClick) {
+      onClick(ev);
+    } else if (!onClick && locationHref) {
+      // If no onClick Function was provided and we do have an locationHref, redirect to the locationHref
+      window.location.href = locationHref;
```

Comment:
this is probably a bad idea. you should let the default behavior go through. That way if the user middle clicks or control clicks on the anchor, it will open in a new window. as an example.